app/jobs/elasticsearch_delete_by_query_job.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+# Deletes documents from the Elasticsearch index where "updated_at" field is older than the configured retention days
+# - Uses async delete_by_query and polls task progress
+# - Configurable throttling and defaults
+
+class ElasticsearchDeleteByQueryJob < ApplicationJob
+  queue_as :searchgov
+
+  DEFAULT_REQUESTS_PER_SECOND = 500
+  DEFAULT_SCROLL_SIZE = 5000
+  DEFAULT_MAX_TASK_WAIT_SECONDS = 23 * 60 * 60 # 23h
+  START_RETRY_ATTEMPTS = 3
+
+  def perform
+    index = fetch_env!('SEARCHELASTIC_INDEX')
+    retention_days = fetch_retention_days!
+
+    Rails.logger.info { "ElasticsearchDeleteByQueryJob starting for index=#{index}, retention_days=#{retention_days}" }
+
+    body = {
+      query: {
+        range: {
+          updated_at: { lt: "now-#{retention_days}d/d" }
+        }
+      }
+    }
+
+    requests_per_second = DEFAULT_REQUESTS_PER_SECOND
+    scroll_size = DEFAULT_SCROLL_SIZE
+    max_wait_seconds = DEFAULT_MAX_TASK_WAIT_SECONDS
+
+    slices = 'auto'
+
+    response = with_start_retries(index, body, slices, requests_per_second, scroll_size)
+
+    task_id = (response.is_a?(Hash) && response['task']) ? response['task'] : nil
+
+    if task_id.nil?
+      deleted = (response && response['deleted']) || 0
+      Rails.logger.info { "delete_by_query completed synchronously; deleted=#{deleted} index=#{index}" }
+      return
+    end
+
+    Rails.logger.info { "delete_by_query started async task_id=#{task_id} index=#{index} slices=#{slices} rps=#{requests_per_second}" }
+
+    begin
+      result = wait_for_task_completion(task_id, max_wait_seconds)
+      if result[:completed]
+        deleted = result[:deleted] || 0
+        Rails.logger.info { "Async delete_by_query finished for index=#{index}: deleted=#{deleted}" }
+      else
+        Rails.logger.warn { "Async delete_by_query for index=#{index} did not finish within #{max_wait_seconds}s; task_id=#{task_id}" }
+        ES.client.tasks.cancel(task_id: task_id)
+      end
+    rescue Elasticsearch::Transport::Transport::Errors::NotFound => e
+      Rails.logger.error { "Task not found while monitoring delete_by_query for index=#{index}: #{e.message}" }
+      raise
+    rescue StandardError => e
+      Rails.logger.error { "Unexpected error while monitoring delete_by_query for index=#{index}: #{e.message}" }
+      raise
+    end
+  end
+
+  private
+
+  def fetch_env!(key)
+    ENV.fetch(key) { raise KeyError, "#{key} must be set in environment" }
```

Comment:
Env.fetch raises an error already if the key does't exits. No need to setup another raise condition. Just do `ENV.fetch(key)`